File: 2025_07/leetcode2025-07-19.py
from typing import List
from collections import defaultdict

class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort()
        cnt = defaultdict(int)
        for i in folder:
            cur = i.split('/')
            flag = True
            for j in range(len(cur)):
                if '/'.join(cur[:j+1]) in cnt:
                    flag = False
                    break
            if flag:
                cnt[i] += 1
        return list(cnt.keys())

if __name__ == "__main__":
    s = Solution()
    folder = ["/a","/a/b","/c/d","/c/d/e","/c/f"]
    print(s.removeSubfolders(folder))

File: 2025_07/leetcode2025-07-19.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        sort(folder.begin(), folder.end());
        vector<string> res = {folder[0]};
        for (int i=1; i<folder.size(); i++){
            string cur = folder[i], last = res[res.size()-1];
            int m = last.size(), n = cur.size();
            if (m>=n || !(last==cur.substr(0, m) && cur[m]=='/')){
                res.push_back(cur);
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<string> folder = {"/a","/a/b","/c/d","/c/d/e","/c/f"};
    vector<string> res = s.removeSubfolders(folder);
    for (auto&i : res){
        cout << i << " ";
    }
    cout << endl;
}
